src/codeu/chat/common/UserType.java

```diff
@@ -0,0 +1,49 @@
+package codeu.chat.common;
+
+import codeu.chat.util.Serializer;
+import codeu.chat.util.Serializers;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public enum UserType {
+  CREATOR(0),
+  OWNER(1),
+  MEMBER(2),
+  NOTSET(3);
+
+  public final int fId;
+
+  UserType(int id) {
+    fId = id;
+  }
+
+  public static final Serializer<UserType> SERIALIZER =
+      new Serializer<UserType>() {
+        @Override
+        public void write(OutputStream out, UserType value) throws IOException {
+          Serializers.INTEGER.write(out, value.fId);
+        }
+
+        @Override
+        public UserType read(InputStream in) throws IOException {
+          return UserType.fromId(Serializers.INTEGER.read(in));
+        }
+      };
+
+  public static UserType fromId(int id) {
+    return values()[id];
```

Comment:
I think using the int makes sense for serialization. In other cases it's unnecessary, since using the enum name is preferable. It's similar to preferring named constants over "magic numbers."

`levelCompare` is actually a bigger issue, since it's very brittle. Let's say you have too many `MEMBER` users and want to throttle their messages. But you like some members, so you want to make a user type with unrestricted messages. This new type, `ELITE_MEMBER`, would be between `MEMBER` and `OWNER`. Since there's no int between 2 and 3, you have to refactor every use of `levelCompare` first. As the code base gets bigger, such a refactor becomes more painful.